packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,248 @@
+    data: ConflictDetectionConfig
+  ): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('conflict_detect_convert_org_source'))
+      .withArg('force:mdapi:convert')
+      .withFlag('--rootdir', this.relativeUnpackagedPath)
+      .withFlag('--outputdir', this.relativeConvertedPath)
+      .withLogName('conflict_detect_convert_org_source')
+      .build();
+  }
+
+  public async checkForConflicts(
+    data: ConflictDetectionConfig
+  ): Promise<DirectoryDiffResults> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const results = await this.checkForConflictsInternal(
+      getRootWorkspacePath(),
+      data,
+      cancellationTokenSource,
+      cancellationToken
+    );
+    return results;
+  }
+
+  private async checkForConflictsInternal(
+    projectPath: string,
+    data: ConflictDetectionConfig,
+    cancellationTokenSource: any,
+    cancellationToken: any
+  ): Promise<DirectoryDiffResults> {
+    const projectMetadataTempPath = path.join(
+      projectPath,
+      this.relativeMetdataTempPath
+    );
+    const retrievePackageXmlPath = path.join(
+      projectPath,
+      this.relativePackageXmlPath
+    );
+    const remoteSourcePath: string = path.join(
+      projectPath,
+      this.relativeConvertedPath
+    );
+    const unpackagedZipFile = path.join(
+      projectMetadataTempPath,
+      'unpackaged.zip'
+    );
+    const localSourcePath: string = path.join(projectPath, data.packageDir);
+
+    // 1: create the shadow directory
+    try {
+      shell.mkdir('-p', projectMetadataTempPath);
+      shell.cp(data.manifest, retrievePackageXmlPath);
+    } catch (error) {
+      console.error(error);
+      channelService.appendLine(
+        nls.localize('error_creating_packagexml', error.toString())
+      );
+      notificationService.showErrorMessage(
+        nls.localize('error_creating_packagexml', error.toString())
+      );
+      return Promise.reject();
+    }
+
+    // 2: retrieve unmanaged org source to the shadow directory
+    await this.executeCommand(
+      this.buildRetrieveOrgSourceCommand(data),
+      { cwd: projectPath },
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 3: unzip retrieved source
+    try {
+      const zip = new AdmZip(unpackagedZipFile);
+      zip.extractAllTo(projectMetadataTempPath, true);
+    } catch (error) {
+      console.error(error);
+      channelService.appendLine(
+        nls.localize('error_extracting_org_source', error.toString())
+      );
+      notificationService.showErrorMessage(
+        nls.localize('error_extracting_org_source', error.toString())
+      );
+      return Promise.reject();
+    }
+
+    // 4: convert org source to decomposed (source) format
+    await this.executeCommand(
+      this.buildMetadataApiConvertOrgSourceCommand(data),
+      { cwd: projectPath },
```

Comment:
We should add `env: { SFDX_JSON_TO_STDOUT: 'true' }` here so that all error messages are surfaced through stdout since them being surfaced through stderr will be deprecated. 
https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_cli_env_variables.htm

Changing this will mean reworking a bit what's on `attachExecution`.

Reply:
Let me correct myself, stderr surfacing errors won't be deprecated but it won't be the default behavior in the near future.